src/main/java/org/broadinstitute/hellbender/tools/examples/metrics/multi/ExampleMultiMetricsCollector.java

```diff
@@ -0,0 +1,144 @@
+                    Integer,
+                    PerUnitExampleMultiMetricsCollectorArgs,
+                    PerUnitExampleMultiMetricsCollector>
+        implements Serializable
+{
+    private static final long serialVersionUID = 1L;
+    private static final Logger log = LogManager.getLogger(ExampleMultiMetricsCollector.class);
+
+    ExampleMultiMetricsArgumentCollection inputArgs = null;
+
+    /**
+     * @param inputArgs ExampleMultiMetricsArgumentCollection populated with argument values. May not be null.
+     * @param samHeader samHeader for the input to be processed. May not be null.
+     */
+    public ExampleMultiMetricsCollector(
+            final ExampleMultiMetricsArgumentCollection inputArgs,
+            final SAMFileHeader samHeader)
+    {
+        Utils.nonNull(inputArgs);
+        Utils.nonNull(samHeader);
+
+        this.inputArgs = inputArgs;
+
+        // call setup to create the level distributors
+        setup(inputArgs.metricAccumulationLevel, samHeader.getReadGroups());
+    }
+
+    /**
+     * Return the read filter for example metrics collector.
+     * @param samHeader SAMFileHeader for the input file. May not be null
+     * @return ReadFilter to be used to filter records
+     */
+    public ReadFilter getReadFilter(final SAMFileHeader samHeader) {
+        Utils.nonNull(samHeader);
+        return ReadFilterLibrary.ALLOW_ALL_READS; // no filtering for this example
+    }
+
+    /**
+     * Process the record and create a collector args object to hold the results of any
+     * calculation for re-use across multiple levels.
+     */
+    @Override
+    protected PerUnitExampleMultiMetricsCollectorArgs makeArg(
+            final SAMRecord samRecord,
+            final ReferenceSequence refSeq) {
+        // Example collector doesn't use any cached values from the record so its a no-op.
+        return new PerUnitExampleMultiMetricsCollectorArgs();
+    }
+
+    /**
+     * Make an PerUnitExampleMultiMetricsCollector with the given arguments
+     */
+    @Override
+    protected PerUnitExampleMultiMetricsCollector makeChildCollector(
+            final String sample,
+            final String library,
+            final String readGroup) {
+        return new PerUnitExampleMultiMetricsCollector(
+                sample,
+                library,
+                readGroup);
+    }
+
+    /**
+     * Combine two ExampleMultiMetricsCollector objects and return a single ExampleMultiMetricsCollector
+     * object representing the combined results. NOTE: this implementation is destructive in that it
+     * merges the source into the target and returns the target as the combined object.
+     * @param target target destination of combined metrics. May not be null.
+     * @param source source of metrics to be combined into target. May not be null.
+     * @return single object representing the combined source and target objects
+     */
+    public ExampleMultiMetricsCollector combine(
```

Comment:
Good point about the private/protected.

I think the compiler is getting confused because there's an overload of it.
```
reduce(ExampleMultiMetricsCollector::combine)
```
blows up, but
```
.reduce((a,b) -> ExampleMultiMetricsCollector.combine(a,b)); 
``` 
Works for me.